### ISSUE TITLE
Sleef_cosf1_u35purec returns infinity for -90°

### ISSUE THREAD

**Issue by WSSDude** [shibatch/sleef#276]
As title says. I believe this should not happen.

I tested out this little snippet to assure myself there is nothing wrong with my code. u10 version seems to work as expected (kinda... seems weird it outputs -0 instead of 0 but that is not an issue for me), sin seems to be not affected in whichever version (with shifted value of course).

There also seems to be variation between debug and release builds output. Just thought I might point this out too.

Test was done with following snippet:
```c
#include <sleef.h>
#include <stdio.h>

#define DEG2RAD 1.745329238474369049072265625e-2f

int main() {
  printf("cos_u10(-90*pi/180) = %f\n", Sleef_cosf1_u10purec(-90.0f*DEG2RAD));
  printf("cos_u10( 90*pi/180) = %f\n", Sleef_cosf1_u10purec( 90.0f*DEG2RAD));
  printf("cos_u35(-90*pi/180) = %f\n", Sleef_cosf1_u35purec(-90.0f*DEG2RAD));
  printf("cos_u35( 90*pi/180) = %f\n", Sleef_cosf1_u35purec( 90.0f*DEG2RAD));
  
  printf("sin_u10(-0 *pi/180) = %f\n", Sleef_sinf1_u10purec(-0.0f *DEG2RAD));
  printf("sin_u10( 0 *pi/180) = %f\n", Sleef_sinf1_u10purec( 0.0f *DEG2RAD));
  printf("sin_u35(-0 *pi/180) = %f\n", Sleef_sinf1_u35purec(-0.0f *DEG2RAD));
  printf("sin_u35( 0 *pi/180) = %f\n", Sleef_sinf1_u35purec( 0.0f *DEG2RAD));

  return 0;
}
```

Output (debug build):
```
cos_u10(-90*pi/180) = -0.000000
cos_u10( 90*pi/180) = -0.000000
cos_u35(-90*pi/180) = 1.#INF00
cos_u35( 90*pi/180) = -0.000000
sin_u10(-0 *pi/180) = -0.000000
sin_u10( 0 *pi/180) = 0.000000
sin_u35(-0 *pi/180) = -0.000000
sin_u35( 0 *pi/180) = 0.000000
```

Output (release build):
```
cos_u10(-90*pi/180) = 0.000000
cos_u10( 90*pi/180) = -0.000000
cos_u35(-90*pi/180) = 1.#INF00
cos_u35( 90*pi/180) = -0.000000
sin_u10(-0 *pi/180) = 0.000000
sin_u10( 0 *pi/180) = 0.000000
sin_u35(-0 *pi/180) = -0.000000
sin_u35( 0 *pi/180) = 0.000000
```

**Comment by WSSDude** [comment]
Little addendum - output from vectorized versions seems to be as expected for more precise version and less precise (although same inconsistency with signs can be observed between different builds).

**Comment by shibatch** [comment]
Hello,

Thank you for reporting an issue.
It does not reproduce on my environment.
Please tell us your compiler version, OS version, etc.

```
[namihei]~/work/sleef3/sleef-master/build-x86_64$ cat bug.c
#include <sleef.h>
#include <stdio.h>

#define DEG2RAD 1.745329238474369049072265625e-2f

int main() {
  printf("cos_u10(-90*pi/180) = %f\n", Sleef_cosf1_u10purec(-90.0f*DEG2RAD));
  printf("cos_u10( 90*pi/180) = %f\n", Sleef_cosf1_u10purec( 90.0f*DEG2RAD));
  printf("cos_u35(-90*pi/180) = %f\n", Sleef_cosf1_u35purec(-90.0f*DEG2RAD));
  printf("cos_u35( 90*pi/180) = %f\n", Sleef_cosf1_u35purec( 90.0f*DEG2RAD));

  printf("sin_u10(-0 *pi/180) = %f\n", Sleef_sinf1_u10purec(-0.0f *DEG2RAD));
  printf("sin_u10( 0 *pi/180) = %f\n", Sleef_sinf1_u10purec( 0.0f *DEG2RAD));
  printf("sin_u35(-0 *pi/180) = %f\n", Sleef_sinf1_u35purec(-0.0f *DEG2RAD));
  printf("sin_u35( 0 *pi/180) = %f\n", Sleef_sinf1_u35purec( 0.0f *DEG2RAD));

  return 0;
}
[namihei]~/work/sleef3/sleef-master/build-x86_64$ gcc bug.c -I./include -L./lib -lsleef
[namihei]~/work/sleef3/sleef-master/build-x86_64$ LD_LIBRARY_PATH=./lib ./a.out
cos_u10(-90*pi/180) = -0.000000
cos_u10( 90*pi/180) = -0.000000
cos_u35(-90*pi/180) = -0.000000
cos_u35( 90*pi/180) = -0.000000
sin_u10(-0 *pi/180) = -0.000000
sin_u10( 0 *pi/180) = 0.000000
sin_u35(-0 *pi/180) = -0.000000
sin_u35( 0 *pi/180) = 0.000000
```

**Comment by WSSDude** [comment]
Oh, I'm sorry. I used my version from my build tree, which had some fixes for VS2017 and VS2019 (as I was not able to compile it there properly within my build tree) and changes in CMake files that I needed to incorporate because of my particular usage.

Anyway, with clean source from here and minimal modifications in CMake files needed to build it within my build tree, it seems to work properly now.
After this, I revisited my changes I did in my fork (which I thought were minor) and found out one messed up line that I changed. After fixing this line with proper fix and not one I made, this little test seems to run as expected.

So false alarm, this was solely my bad in my fork, not here as I thought. I'll try to be more thorough next time before reporting something again, sorry.
